app/routes/circulars/circulars.server.ts

```diff
@@ -263,3 +270,59 @@ export async function circularRedirect(query: string) {
     throw redirect(circularURL)
   }
 }
+
+async function* getAllRecords(): AsyncGenerator<Circular[], void, unknown> {
+  const db = await tables()
+  const client = db._doc as unknown as DynamoDBDocument
+  const TableName = db.name('circulars')
+  const pages = paginateScan({ client }, { TableName })
+
+  for await (const page of pages) {
+    const items: Circular[] = page.Items as Circular[]
+    yield items
+  }
+}
+
+export async function makeTarFile(
+  fileType: 'json' | 'txt'
+): Promise<ReadableStream> {
+  const tarStream = new Readable({
+    read() {},
+  })
+
+  const pack = tarPack()
+
+  pack.on('error', (err: Error) => {
+    tarStream.emit('error', err)
+  })
+
+  pack.on('data', (chunk: Uint8Array) => {
+    tarStream.push(chunk)
+  })
+
+  pack.on('end', () => {
+    tarStream.push(null)
+  })
+  for await (const circularArray of getAllRecords()) {
+    for (const circular of circularArray) {
+      if (fileType === 'txt') {
+        const txt_entry = pack.entry(
+          { name: `archive/${circular.circularId}.txt` },
```

Comment:
The directory should match the name of the archive.

```suggestion
          { name: `archive.txt/${circular.circularId}.txt` },
```